charts/opentelemetry-demo-webstore/values.yaml

```diff
@@ -0,0 +1,137 @@
+observability:
+  otelcol:
+    enabled: true
+  jaeger:
+    enabled: true
+
+image:
+  repository: otel/demo
+  pullSecrets: []
+
+serviceAccount: ""
+
+components:
+  redis:
+    enabled: true
+    image: redis:alpine
+    ports:
+    - name: redis
+      value: 6379
+
+  ad-service:
+    enabled: true
+    servicePort: 8080
+
+  cart-service:
+    enabled: true
+    servicePort: 8080
+    env:
+    - name: ASPNETCORE_URLS
+      value: http://*:8080
+    depends_on:
```

Comment:
https://github.com/open-telemetry/opentelemetry-helm-charts/pull/280/files#diff-e02a306853f89e21f54ac77efe8a5a03f1f4df85f9d316a58f15a2a578d51382R22-R28

The inspiration here is the docker compose file from otel-demo.
For some reasons, the component needs to manually declare the endpoint of the dependent service via environment variables.

1. For example, the front service depends on shipping. 
If the shipping port is modified, you need to modify the environment variables of multiple services.
With depends_on, you only need to modify one port, and the other environment variables are automatically generated with the correct names.

2. Our k8s service includes the release name, so the pre-set environment variables should not include the release name.

Reply:
I understand the meaning of `depends_on`, I just don't understand why it has to be in values.yaml - why a helm chart user would need to change the `depend_on` list?

Reply:
In general, users do not need to modify this depends on.
The default value in value.yaml is sufficient.
This way of generating all yaml in bulk is a bit special compared to the compoent one yaml file approach. But I think it's worth it because we can generate yaml for a dozen components with a small amount of code and declarations

Reply:
If users shouldn't be changing the items in `depends_on`, then we should probably move it to a helper template.  I think we'd be able to get the same functionality that way but it wouldn't need to be in values.yaml.

Reply:
move depends logic  to _pod.tpl, thx suggest